src/syscheckd/create_db.c

```diff
@@ -429,7 +429,11 @@ static int read_file(const char *file_name, int dir_position, whodata_evt *evt,
                 free(s_node);
                 merror("Unable to add file to db: %s", file_name);
             }
-
+#ifndef WIN32
+            if (OSHash_Add_ex(syscheck.inode_hash, str_inode, strdup(file_name)) <= 0) {
```

Comment:
The error condition is **!= 2**, not <= 0.

Reply:
strdup reserves memory whose reference is lost if OSHash_Add_ex fails. You can use an auxiliary variable to release it in such case.